### PR TITLE
HYDRATOR-699 remove etl batch from tests

### PR DESCRIPTION
**Pull request by albertshau**
Moving tests from etl-batch to data-pipeline

### CODE REVIEW

**Comment by albertshau** [comment]
http://builds.cask.co/browse/CDAP-RUT305

File: cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/DataPipelineTest.java [hunk 1]
```diff
@@ -1304,6 +1308,222 @@ private void testSecureStorePipeline(Engine engine, String prefix) throws Except
     Assert.assertNotNull(getDataset(prefix + "MockSecureSinkDataset").get());
   }
 
+  @Test
+  public void testExternalDatasetTrackingMR() throws Exception {
```
Comment (albertshau) [inline cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/DataPipelineTest.java:1312]: these tests were straight cut and paste from etl-batch into data-pipeline, only difference being the workflow name.

File: cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/etl/batch/ETLSparkTestRun.java [hunk 2]
```diff
@@ -1,102 +0,0 @@
...
-      .addStage(new ETLStage("sink1", MockSink.getPlugin("sparkoutput1")))
-      .addStage(new ETLStage("sink2", MockSink.getPlugin("sparkoutput2")))
-      .addStage(new ETLStage("filter", StringValueFilterTransform.getPlugin("name", "samuel")))
-      .addConnection("source", "sink1")
-      .addConnection("source", "filter")
-      .addConnection("filter", "sink2")
-      .build();
-
-    AppRequest<ETLBatchConfig> appRequest = new AppRequest<>(APP_ARTIFACT, etlConfig);
-    ApplicationId appId = NamespaceId.DEFAULT.app("DagApp");
```
Comment (CuriousVini) [inline cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/etl/batch/ETLSparkTestRun.java:69]: We dont need this test in dataPipelineTest?

File: cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/etl/batch/ETLBatchTestSuite.java [hunk 3]
```diff
@@ -1,36 +0,0 @@
...
-import co.cask.cdap.test.XSlowTests;
-import org.junit.experimental.categories.Category;
-import org.junit.runner.RunWith;
-import org.junit.runners.Suite;
-
-/**
- * This is a test suite that runs all tests in for ETL batch. This avoids starting/stopping the unit-test framework
- * for every test class.
- */
-@Category(XSlowTests.class)
```
Comment (CuriousVini) [inline cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/etl/batch/ETLBatchTestSuite.java:29]: Why was this test suite marked as slow tests? Should we use this category for some tests from `ETLWorkflowTestRun` and `ETLSparkTestRun`

File: cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/etl/batch/ETLBatchTestSuite.java [hunk 4]
```diff
@@ -1,36 +0,0 @@
...
-@RunWith(TestSuite.class)
-@Suite.SuiteClasses({
-  ETLWorkflowTestRun.class,
```
Comment (CuriousVini) [inline cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/etl/batch/ETLBatchTestSuite.java:32]: Should we also remove `ETLWorkflowTestRun.java`?

File: cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/DataPipelineTest.java (hunk 1 above)
Comment (CuriousVini) [reply cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/DataPipelineTest.java:1312]: Shouldn't we include `ETLWorkflowTestRun#testInvalidTransformConfigFailsToDeploy` test?

Comment (albertshau) [reply cdap-app-templates/cdap-etl/cdap-data-pipeline/src/test/java/co/cask/cdap/datapipeline/DataPipelineTest.java:1312]: That test is not a good test. The code path is covered by the tests for the config class itself and should not need a very expensive test that builds jars and tries to deploy an app.

File: cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/etl/batch/ETLSparkTestRun.java (hunk 2 above)
Comment (albertshau) [reply cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/etl/batch/ETLSparkTestRun.java:69]: this seemed to be covered by existing tests in DataPipeline so I didn't move it.

File: cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/etl/batch/ETLBatchTestSuite.java (hunk 4 above)
Comment (albertshau) [reply cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/etl/batch/ETLBatchTestSuite.java:32]: hm meant to remove that class completely. will update.

**Comment by CuriousVini** [comment]
lgtm

**Pull request merged**